### ISSUE TITLE
Implement regrid operation based on datashader

### ISSUE THREAD

**Issue by philippjfr** [ioam/holoviews#1552]
In https://github.com/bokeh/datashader/pull/375 a fast regridding implementation with no additional dependencies was merged into datashader. Since we already provide an ``aggregate`` operation we should simply add a ``regrid`` operation to the operation.datashader module in HoloViews, which will allow for much faster regridding of ``Image`` types than the current approach which expands the Image to a large table and then runs aggregation on it. I have already written an operation based on the original gridtools library the datashader implementation was based on, which should be easy to adapt to use the datashader API instead.

**Comment by philippjfr** [comment]
This is now waiting on various issues in datashader to be addressed (https://github.com/bokeh/datashader/issues/380), I've written the operation now but there are a number of showstoppers before I can continue.

**Comment by philippjfr** [comment]
Reassigning to 1.9 since this relies on changes in datashader.

**Comment by philippjfr** [comment]
The regrid operation has now been implemented.
